common/core/desktop/tests/unit/kmx/kmx.cpp

```diff
@@ -313,10 +330,14 @@ int run_test(const km::kbp::path & source, const km::kbp::path & compiled) {
 
   // Run through key events, applying output for each event
   for (auto p = next_key(keys); p.vk != 0; p = next_key(keys)) {
-    try_status(km_kbp_process_event(test_state, p.vk, p.modifier_state));
-
-    for (auto act = km_kbp_state_action_items(test_state, nullptr); act->type != KM_KBP_IT_END; act++) {
-      apply_action(test_state, *act, text_store, options);
+    if (p.vk == KM_KBP_VKEY_CAPS) {
+      toggle_caps_lock_state();
+    }
```

Comment:
```suggestion
    // Because a normal system tracks caps lock state itself,
    // we mimic that in the tests. We assume caps lock state is
    // updated on key_down before the processor receives the
    // event.
    if (p.vk == KM_KBP_VKEY_CAPS) {
      toggle_caps_lock_state();
    }
```